src/states/adult.rs

```diff
@@ -436,11 +436,22 @@ impl Approved for Adult {
         false
     }
 
-    fn handle_add_elder_event(
+    fn handle_online_event(
         &mut self,
-        pub_id: PublicId,
+        payload: OnlinePayload,
         outbox: &mut dyn EventBox,
     ) -> Result<(), RoutingError> {
+        let pub_id = *payload.p2p_node.public_id();
+
+        if !self.chain.can_add_member(&payload.p2p_node) {
+            info!("{} - ignore Online: {:?}.", self, payload);
+            return Ok(());
+        }
+
+        info!("{} - handle Online: {:?}.", self, payload);
+        self.chain.add_member(payload.p2p_node, payload.age);
```

Comment:
How did you make it so the new nodes got gossip? Is it that they get the gossip from the new parsec that they are part of? but then did we not need to change the membership in NodeApproval?

Reply:
This actually works without modifications, even though it's suboptimal. The new node basically gets a genesis pfx info for the last Parsec before it joined - it pokes the existing nodes and receives all the events from the old Parsec, then it's able to analyse them, get consensus about the new SectionInfo and generate the new genesis pfx info by itself.

We could send it the new genesis pfx info, but it would have to be done only after the SectionInfo accumulates, as that is when we actually generate it. I'm not sure if this is safe with the current flow.

Reply:
That is weird, I thought it would not get gossip from old parsec since it was never added to it. Did I miss something?

Reply:
Hmm, that is a good question. I did think that something would break, but nothing did. Maybe it's mock_parsec not simulating this properly? I'm not sure.
Or the new node can somehow get gossip, because it's added to the new Parsec? But that would be weird, too.

Reply:
Yes mock parsec does:
```
    pub fn create_gossip(&self, _peer_id: &S::PublicId) -> Result<Request<T, S::PublicId>, Error> {
        Ok(Request::new())
    }
```

And so there is not check there.

Reply:
This can be solved with follow up commits that I've been working on and that start DKG (thus adding new node to gossip) mock_base seem to be passing with it.